app/lib/widgets/wallets/activate_wallet.dart

```diff
@@ -0,0 +1,355 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:threebotlogin/helpers/globals.dart';
+import 'package:threebotlogin/helpers/logger.dart';
+import 'package:threebotlogin/models/wallet.dart';
+import 'package:threebotlogin/providers/wallets_provider.dart';
+import 'package:threebotlogin/services/stellar_service.dart' as StellarService;
+import 'package:threebotlogin/widgets/custom_dialog.dart';
+
+class ActivateWalletWidget extends ConsumerStatefulWidget {
+  const ActivateWalletWidget({super.key, required this.wallet});
+  final Wallet wallet;
+
+  @override
+  ConsumerState<ActivateWalletWidget> createState() =>
+      _ActivateWalletWidgetState();
+}
+
+class _ActivateWalletWidgetState extends ConsumerState<ActivateWalletWidget> {
+  Wallet? _selectedWallet;
+  String? walletError;
+  bool saveLoading = false;
+  late WalletsNotifier walletRef;
+  int tftPrice = 0;
+  bool isLoadingPrice = true;
+
+  @override
+  void initState() {
+    super.initState();
+    walletRef = ref.read(walletsNotifier.notifier);
+    _loadTFTPrice();
+  }
+
+  Future<void> _loadTFTPrice() async {
+    setState(() {
+      isLoadingPrice = true;
+    });
+
+    try {
+      final price = await StellarService.getTFTPriceFromXLM();
+      if (mounted) {
+        setState(() {
+          tftPrice = price;
+          isLoadingPrice = false;
+        });
+      }
+    } catch (e) {
+      logger.e('Failed to load TFT price: $e');
+      if (mounted) {
+        setState(() {
+          tftPrice = 3;
+          isLoadingPrice = false;
+        });
+      }
+    }
+  }
+
+  List<DropdownMenuEntry<Wallet>> _buildDropdownMenuEntries(
+      List<Wallet> wallets) {
+    return wallets
+        .where((wallet) =>
+            wallet != widget.wallet && wallet.stellarBalance != '-1')
+        .map((wallet) {
+      return DropdownMenuEntry<Wallet>(
+        value: wallet,
+        label: wallet.name,
+        labelWidget: Text(wallet.name,
+            style: Theme.of(context).textTheme.bodyLarge!.copyWith(
+                  color: Theme.of(context).colorScheme.onSurface,
+                )),
+      );
+    }).toList();
+  }
+
+  bool _validateWallet() {
+    if (_selectedWallet == null) {
+      setState(() {
+        walletError = 'Please select a wallet';
+      });
+      return false;
+    }
+    if (double.parse(_selectedWallet!.stellarBalance) < (tftPrice * 3)) {
+      setState(() {
```

Comment:
Can we use a constant instead of hardcoding the value 3, just to make things clearer?